OpenOversight/app/main/views.py

```diff
@@ -74,26 +80,49 @@ def submit_complaint():
                            officer_image=request.args.get('officer_image'))
 
 
-@main.route('/submit')
+@main.route('/submit', methods=['GET', 'POST'])
+@login_required
 def submit_data():
-    return render_template('submit.html')
-
-
-@main.route('/upload', methods=['POST'])
-def upload_file():
-    file = request.files['file']
-    if file and allowed_file(file.filename):
-        filename = secure_filename(file.filename)
-        file.save(os.path.join(current_app.config['UNLABELLED_UPLOADS'], filename))
-        return redirect(url_for('main.show_upload',
-                                filename=filename))
-
-
-@main.route('/show_upload/<filename>')
-def show_upload(filename):
-    # return send_from_directory('../'+config.UNLABELLED_UPLOADS,
-    #                           filename)
-    return 'Successfully uploaded: {}'.format(filename)
+    form = HumintContribution()
+    if request.method == 'POST' and form.validate_on_submit():
+        original_filename = secure_filename(request.files[form.photo.name].filename)
+        image_data = request.files[form.photo.name].read()
+
+        # See if there is a matching photo already in the db
+        hash_img = compute_hash(image_data)
+        hash_found = Image.query.filter_by(hash_img=hash_img).first()
+
+        if not hash_found:
+            # Generate new filename
+            file_extension = original_filename.split('.')[-1]
```

Comment:
`FileAllowed` in `forms.py` asserts that the file uploaded ends in `{.jpg,.jpeg,.png}` right? Because if not, this could be dicey. Can we throw a test or two in that assert that a file with name ending in .jpg, .jpeg, .png gets accepted and more importantly, one with a name ending in, say, `/../../../../../~/etc/passwd` doesn't, so we capture this requirement in our tests?